packages/jaeger-ui/src/reducers/deep-dependency-graph.tsx

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { handleActions } from 'redux-actions';
+
+import { actionTypes } from '../actions/deep-dependency-graph';
+import { fetchDeepDependencyGraph } from '../actions/jaeger-api';
+import { fetchedState } from '../constants';
+import { ApiError } from '../types/api-error';
+import transformDdgData from '../model/ddg/transformDdgData';
+import {
+  stateKey,
+  TDdgActionMeta,
+  TDdgAddStylePayload,
+  TDdgClearStylePayload,
+  TDdgPayload,
+  TDdgState,
+  TDdgStateEntry,
+} from '../model/ddg/types';
+import guardReducer, { guardReducerWithMeta } from '../utils/guardReducer';
+
+export function addStyleState(state: TDdgState, { payload }: { payload: TDdgAddStylePayload }) {
+  const { service, operation, start, end, visibilityIndices, style } = payload;
+  const key = stateKey(service, operation, start, end);
+  const stateEntry: TDdgStateEntry | undefined = state[key];
+  if (!stateEntry || stateEntry.state !== fetchedState.DONE) {
+    console.warn('Cannot set style state for unloaded Deep Dependency Graph'); // eslint-disable-line no-console
+    return state;
+  }
+  const styleStates = new Map(stateEntry.styleStates);
+  visibilityIndices.forEach(idx => {
+    styleStates.set(idx, (styleStates.get(idx) || 0) | style); // eslint-disable-line no-bitwise
+  });
+  return {
+    ...state,
+    [key]: {
+      ...stateEntry,
+      styleStates,
+    },
+  };
+}
+
+export function clearStyleState(state: TDdgState, { payload }: { payload: TDdgClearStylePayload }) {
```

Comment:
As an aside, it might be handy to use a util class for dealing with view modifers:

```ts
class ViewModifiers extends Map<number, number> {
  clone() {
    return new ViewModifiers(this);
  }

  rmModifier(modifier: EViewModifier | null, index: number) {
    if (modifier == null) {
      this.delete(index);
      return;
    }
    const value = (this.get(index) || 0) & ~modifier;
    if (value === EViewModifier.None) {
      this.delete(index);
    } else {
      this.set(index, value);
    }
  }

  addModifier(modifier: EViewModifier, index: number) {
    const value = (this.get(index) || 0) | modifier;
    this.set(index, value);
  }
}
```

Reply:
Things like `emphasizing` all paths through a vertex will mean that many pathElems need to be added/removed at once. It may be easier to clear `emphasized` from all elems rather than calculate which need to be `unemphasized`
Clearly all for a pathElem could be part of a vertice's `componentWillUnmount` clean up.